packages/shared/src/utils/helpers/format-response.ts

```diff
@@ -67,26 +69,26 @@ export const formatIDVError = (errors: string[], status_code: string) => {
      */
     if (errors.length === 0 && (status_code === STATUS_CODES.NONE || status_code === STATUS_CODES.VERIFIED))
         return null;
-    const error_keys: Record<string, TIDVErrorStatus> = {
-        name: 'POI_NAME_MISMATCH',
-        birth: 'POI_DOB_MISMATCH',
-        rejected: 'POI_FAILED',
-    };
     if (status_code === STATUS_CODES.EXPIRED) {
-        return 'POI_EXPIRED';
+        return IDV_ERROR_STATUS.Expired.code;
     }
-    const status: TIDVErrorStatus[] = [];
+    const status: Array<TIDVErrorStatus> = [];
     errors.forEach(error => {
-        const error_regex = RegExp(/(name|birth|rejected)/i).exec(error);
-        if (error_regex) {
-            status.push(error_keys[error_regex[0].toLowerCase()]);
+        const error_key: TIDVErrorStatus = IDV_ERROR_STATUS[error as keyof typeof IDV_ERROR_STATUS]
+            .code as TIDVErrorStatus;
+        if (error_key) {
+            status.push(error_key);
         }
     });
-    return status.includes(error_keys.name) &&
-        status.includes(error_keys.birth) &&
-        !status.includes(error_keys.rejected)
-        ? 'POI_NAME_DOB_MISMATCH'
-        : status[0] ?? 'POI_FAILED';
+    return status.includes(IDV_ERROR_STATUS.NameMismatch.code as TIDVErrorStatus) &&
+        status.includes(IDV_ERROR_STATUS.DobMismatch.code as TIDVErrorStatus) &&
+        !status.includes(IDV_ERROR_STATUS.Failed.code as TIDVErrorStatus)
+        ? IDV_ERROR_STATUS.NameDOBMismatch.code
+        : status[0] ?? IDV_ERROR_STATUS.Failed.code;
```

Comment:
There is alot of type assertion happening here, would you consider using a typeguard instead of casting in so many places?

Reply:
Typed the constant